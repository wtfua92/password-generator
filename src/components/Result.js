import React from 'react';
import PropTypes from 'prop-types';

function Result({currentPassword}) {
    return (
        <div className="password-generator__result">
            <label htmlFor="password">Your Secure Password: </label>
            <input id="password" type="text" value={currentPassword} disabled />
        </div>
    );
}

Result.propTypes = {
    currentPassword: PropTypes.string
};

export default Result;