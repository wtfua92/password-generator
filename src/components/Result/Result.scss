@import "../../assets/styles/colors";

.password-generator {
  &__result {
    display: flex;
    flex-direction: column;
    padding-bottom: 1.25rem;

    label {
      text-align: center;
      margin-bottom: 1rem;
    }

    input {
      background: #f3f3f3;
      padding: 0.5rem;
      text-align: center;
      line-height: 1.5rem;
      font-size: 1.5rem;
      border-radius: 0.5rem;
      border: none;
      transition: border-bottom-color 500ms ease;
    }

    input.with-password {
      padding: 0.45rem;
      border-bottom: 2px solid $success;
    }
  }
}