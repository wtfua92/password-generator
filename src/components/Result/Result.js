import React from "react";
import PropTypes from "prop-types";

import "./Result.scss";

const Result = ({ currentPassword, savePasswordToClipboard }) => (
  <div className="password-generator__result">
    <label htmlFor="password">Your Secure Password: </label>
    <input
      className={currentPassword ? "with-password" : ""}
      id="password"
      type="text"
      value={currentPassword}
      disabled
    />
    <button
      type="button"
      onClick={() => {
        savePasswordToClipboard(currentPassword);
      }}
      title="Copy to clipboard"
    >
      <img
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABgAAAAYCAYAAADgdz34AAAABmJLR0QA/wD/AP+gvaeTAAAATklEQVRIiWNgGOqAEYn9n4pmwQEThYYSBCxYxLC6hFwwID6AAarEyYD6AAZIjRMUn9PcB6MWjFowagFqTqZqKQoDg6IsoqhUpbkPhj4AABQZBDNxiXEsAAAAAElFTkSuQmCC"
        alt="Copy to clipboard"
      />
    </button>
  </div>
);

Result.defaultProps = {
  currentPassword: "",
  savePasswordToClipboard: () => {}
};

Result.propTypes = {
  currentPassword: PropTypes.string.isRequired,
  savePasswordToClipboard: PropTypes.func.isRequired
};

export default Result;
