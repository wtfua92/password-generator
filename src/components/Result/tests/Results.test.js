import React from "react";
import { render, fireEvent, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import Result from "../Result";

describe("<Result />", () => {
  const password = "123";

  afterEach(cleanup);

  it("should display password", () => {
    const { getByDisplayValue } = render(<Result currentPassword={password} />);
    const input = getByDisplayValue(password);
    expect(input).toBeDefined();
    expect(input).toHaveValue(password);
  });

  it("should copy the password in clipboard on button click", () => {
    const savePasswordToClipboardMock = jest.fn();
    const { getByTitle } = render(
      <Result
        currentPassword={password}
        savePasswordToClipboard={savePasswordToClipboardMock}
      />
    );
    const button = getByTitle("Copy to clipboard");
    fireEvent.click(button);
    expect(savePasswordToClipboardMock).toHaveBeenCalledTimes(1);
    expect(savePasswordToClipboardMock).toHaveBeenCalledWith(password);
  });

  it("should match the snapshot", () => {
    const savePasswordToClipboardMock = jest.fn();
    const { getByDisplayValue } = render(
      <Result
        currentPassword={password}
        savePasswordToClipboard={savePasswordToClipboardMock}
      />
    );
    const input = getByDisplayValue(password);
    expect(input).toMatchSnapshot();
  });
});
