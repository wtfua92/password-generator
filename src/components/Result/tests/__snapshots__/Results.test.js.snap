// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Result /> should match the snapshot 1`] = `
<input
  class="with-password"
  disabled=""
  id="password"
  type="text"
  value="123"
/>
`;
